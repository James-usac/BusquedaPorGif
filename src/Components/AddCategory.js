import { TextField } from '@mui/material'
import React from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
 
    const [inputValue, setInputValue] = React.useState('')

    const handleChange = (event) => {
        setInputValue(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()
        if(inputValue.trim().length > 2) {
            setCategories(prevCategories => [inputValue, ...prevCategories]);
            setInputValue('')
        }   
        console.log("submit Hecho")
    }

    return (
        <form onSubmit={handleSubmit}>
            <TextField 
                label="Buscar"
                variant="outlined"
                value={inputValue}
                onChange={(e) => handleChange(e)}
            />
        </form>
    )
}

//para indicar que el dato es requerido
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
