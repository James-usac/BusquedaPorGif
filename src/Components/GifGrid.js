
import React, { useEffect } from 'react'

export const GifGrid = ({ category }) => {

    const api_key = "CXlwDg9c0PUX5bdLjOUkBPfbZsmWSmAj";

    const [images, setImages] = React.useState([]);

    useEffect(() => {
        getGifs(category);
    }, [category]);

    const getGifs = async (category) => {
        const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=${api_key}`;
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
        setImages(gifs);
        return gifs;
    }

    return (
        <div>
            <h1>{category}</h1>
            {
                images.map(img => (
                    <div key={img.id}>
                        <img src={img.url} alt={img.title} />
                        <br/>
                        <label>{img.title}</label>
                    </div>
                ))
            }
        </div>
    )
}
