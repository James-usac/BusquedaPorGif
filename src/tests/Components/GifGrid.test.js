import { shallow } from "enzyme";
import { GifGrid } from "../../Components/GifGrid";

describe('Pruebas en <GifGrid>', () => {
  
  test('debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifGrid/>);
    expect(wrapper).toMatchSnapshot();
  });
});