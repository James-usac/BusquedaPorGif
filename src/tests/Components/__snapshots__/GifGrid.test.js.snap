// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid> debe de mostrar el componente correctamente 1`] = `
<div>
  <h1 />
  <ForwardRef(ImageList)
    cols={3}
    rowHeight={164}
    sx={
      Object {
        "maxWidth": 500,
      }
    }
  />
  <ForwardRef(Modal)
    aria-describedby="parent-modal-description"
    aria-labelledby="parent-modal-title"
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(Box)
      sx={
        Object {
          "backgroundColor": "white",
          "bgcolor": "background.paper",
          "border": "2px solid #000",
          "boxShadow": 24,
          "left": "50%",
          "pb": 3,
          "position": "absolute",
          "pt": 2,
          "px": 4,
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": 400,
        }
      }
    >
      <h2
        id="parent-modal-title"
      />
      <img
        alt="gif"
        height="200"
        src={null}
        width="400"
      />
    </ForwardRef(Box)>
  </ForwardRef(Modal)>
</div>
`;
